soc/riscv/riscv-privilege/starfive_jh71xx/soc.h

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2020 Cobham Gaisler AB
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __RISCV_VIRT_SOC_H_
+#define __RISCV_VIRT_SOC_H_
+
+#include <soc_common.h>
+#include <devicetree.h>
+
+#define RISCV_MTIME_BASE 0x0200BFF8
+#define RISCV_MTIMECMP_BASE 0x02004000
```

Comment:
Yes, it can be derived from DTS as Linux is doing. but the problem is it's being used by generic timer driver `drivers/timer/riscv_machine_timer.c` and this macro is being defined by other ports as well. Obviously, as long as I keep the name the same, it won't break anything but for the sake of consistency between different ports I was planning to do this improvement in a separate PR.

Reply:
I think this is ok. Other SoC's are doing the same thing atm in `soc.h`. It would be good if it was brought out to DT later, as long as complementary changes can be made for other SoC's as well.